internal/pkg/consensus/election.go

```diff
@@ -168,20 +169,40 @@ func (tm TicketMachine) MakeTicket(ctx context.Context, base block.TipSetKey, ep
 }
 
 // IsValidTicket verifies that the ticket's proof of randomness is valid with respect to its parent.
-func (tm TicketMachine) IsValidTicket(ctx context.Context, base block.TipSetKey,
+func (tm TicketMachine) IsValidTicket(ctx context.Context, base block.TipSetKey, entry *drand.Entry, newPeriod bool,
 	epoch abi.ChainEpoch, miner address.Address, workerSigner address.Address, ticket block.Ticket) error {
-	entropy, err := encoding.Encode(miner)
-	if err != nil {
-		return errors.Wrapf(err, "failed to encode entropy")
-	}
-	randomness, err := tm.chain.SampleChainRandomness(ctx, base, acrypto.DomainSeparationTag_TicketProduction, epoch, entropy)
+	randomness, err := tm.ticketVRFRandomness(ctx, base, entry, newPeriod, miner, epoch)
 	if err != nil {
-		return errors.Wrap(err, "failed to generate epost randomness")
+		return errors.Wrap(err, "failed to generate ticket randomness")
 	}
 
 	return crypto.ValidateBlsSignature(randomness, workerSigner, ticket.VRFProof)
 }
 
+func (tm TicketMachine) ticketVRFRandomness(ctx context.Context, base block.TipSetKey, entry *drand.Entry, newPeriod bool, miner address.Address, epoch abi.ChainEpoch) (abi.Randomness, error) {
+	entropyBuf := bytes.Buffer{}
+	minerEntropy, err := encoding.Encode(miner)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to encode miner entropy")
+	}
+	_, err = entropyBuf.Write(minerEntropy)
+	if err != nil {
+		return nil, err
+	}
+	if !newPeriod { // resample previous ticket and add to entropy
+		ticketSeed, err := tm.sampler.Sample(ctx, base, epoch)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to sample previous ticket")
+		}
+		_, err = entropyBuf.Write(ticketSeed)
```

Comment:
The way that I'm doing this (putting ticket bytes in entropy buffer) doesn't require moving from `SampleChainRandomness` to `Sample`.  Since the closest thing we have to a spec specifies a different ordering of bytes and speccing is not very strong I am very suspicious that the arrangement of bytes will change during interop testing so its possible this gain in flexibility will help us.

I'm fining moving back to `SampleChainRandomness` if reviewer has a strong preference.